### ISSUE TITLE
Flutter 3 : Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null

### ISSUE THREAD

**Issue by fabiancrx** [tomgilder/routemaster#278]
In Flutter 3 the `WidgetsBinding.instance` returns a non-nullable instance, hence accessing it via`WidgetsBinding.instance?.` or `WidgetsBinding.instance!` throws a linting and compilation warning .

Note that this does not prevent the usage of the package with flutter 3.0 but it does shows a few warnings.

<details>
<summary>Sample warning</summary>

```
ERROR: ../../../dev/flutter3/.pub-cache/hosted/pub.dartlang.org/routemaster-1.0.0/lib/routemaster.dart:1209:22: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
ERROR:  - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('../../../dev/flutter3/packages/flutter/lib/src/widgets/binding.dart').
ERROR:       WidgetsBinding.instance!.addPostFrameCallback((_) {
ERROR:                      ^
```

</details>

 

The fix is quite easy, actually its on [this fork](https://github.com/fabiancrx/routemaster/commit/802f2c6d15d387e5aca8b471dcdfb7cc7027218d).

What prevents me from making a quick PR is that merging this code should raise the minimum flutter version to 3.0 , any thoughts ?

**Comment by tomgilder** [comment]
Thanks, I think for the moment I'll add ignores to these lines, feels a little early to require Flutter 3.0

**Comment by fabiancrx** [comment]
Fair enough , the hype was stronger than me 😅. Should I leave the issue open?

**Comment by rich-j** [comment]
The [Flutter 3.0 release notes](https://docs.flutter.dev/development/tools/sdk/release-notes/release-notes-3.0.0) provides details on how to support both Flutter 3 and earlier versions.

Also please apply to the "nested" branch.

**Comment by tomgilder** [comment]
Fixed in 1.0.1, thanks to @josephelliot-wk
